module/VuFind/src/VuFind/IdentifierLinker/BrowZine.php

```diff
@@ -100,11 +102,28 @@ protected function arrayKeyAvailable(string $key, ?array $data): bool
      */
     protected function processServiceLink(array $data, string $serviceKey, array $config): array
     {
+        $serviceData = $data[$serviceKey];
         $result = [
-            'link' => $data[$serviceKey],
-            'label' => $this->translate($config['linkText']),
+            'link' => $serviceData,
             'data' => $data,
         ];
+
+        // If this link is actually the 'bestIntegratorLink' array, extract the appropriate
+        // text and icon config from it.
+        if ('bestIntegratorLink' == $serviceKey) {
+            $result['link'] = $serviceData['bestLink'];
+
+            $linkType = $serviceData['linkType'];
+            $specificConfig = $this->getBestIntegratorLinks()[$linkType] ?? false;
+            if ($specificConfig) {
+                $config = $specificConfig;
+            }
+            if ($this->config['useBrowzineLabel'] ?? false) {
+                $config['linkText'] = $serviceData['recommendedLinkText'];
```

Comment:
Should we also check that `$serviceData['recommendedLinkText']` is non-empty for good measure?